### ISSUE TITLE
RedundantSort rule not working properly in case of max_by(&:size)

### ISSUE THREAD

**Issue by Mifrill** [rubocop/rubocop#10061]
The goal:

`return the last of most sized words in the string`

A possible way to solve it:

```ruby
"hello world".split(/\W+/).sort_by(&:size).last
# => "world"
```

This code raise `rubocop` offense:

```
RuboCop: Use `max_by` instead of `sort_by...last`. [Style/RedundantSort]
```

If we apply the suggested replacement, then we got a different result:

```ruby
"hello world".split(/\W+/).max_by(&:size)
# => "hello"
```

--------

## Expected behavior

No suggest replacement when the result going to change.

## Actual behavior

Suggest replacement according to `Style/RedundantSort` rule with result changes.

## RuboCop version

```
1.18.4 (using Parser 3.0.2.0, rubocop-ast 1.9.0, running on ruby 3.0.2 x86_64-linux)
  - rubocop-rails 2.11.3
  - rubocop-rspec 2.4.0
```

**Comment by dvandersluis** [comment]
Isn't this a problem in general when there are multiple values that translate to the same value? Even without `size`, `sort_by.last` will take the last one and `max_by` will take the first one when there are multiple same values.

I think this cop needs to be marked as unsafe.
